### ISSUE TITLE
Lack of data and features in Chrome.tabs API

### ISSUE THREAD

**Issue by yankovichv** [hober/web-extensions#412]
We are currently working on a tab manager and have encountered a number of limitations in the API.

1. API Chrome.tabs does not return data about whether there is a record in the tab (https://gang.manganum.app/hLmwdsD3). To do this, you need to use chrome.tabCapture, which is absolutely redundant just to display the capture status.

2. The ability to mute a tab is different from muting a browser. It seems to me that extensions should have access to the browser's "Mute site" feature.

There are two kinds of mute: the tab mute and the domain mute. Extensions can set the tab mute, but not the domain mute.
In contrast, the zoom API allows to set both the tab zoom and the domain zoom.

Another problem with this is that the "Mute/unmute site" option in the tab's context menu doesn't work once an extension sets the mute state of a tab. This happens because the tab mute takes precedence over the domain mute.
And, what's worse, once you are in this situation, there's no way for an extension to remove the tab mute from that tab. Even if the extension sets mute=false, the domain mute will still not work.

In summary:
— The user can only control the domain mute.
— The extension can only control the tab mute.
— The tab mute takes precedence over the domain mute.
— Once the tab mute has been set, it cannot be removed.

**Comment by patrickkettner** [comment]
[original chromium-extensions thread](https://groups.google.com/a/chromium.org/g/chromium-extensions/c/brg6LX0baHk/m/eUp4TuD-BQAJ)

**Comment by dotproto** [comment]
Quick notes from today's meeting. The issue report raises two primary issues in a numbered list. The first item (tab's media capture state) seems relevant to all browsers and we're interested in discussing further. The second item (tab mute vs. "browser" mute) seems to be a Chromium-specific issue that should be addressed.

Regarding a tab's current media capture state, @Rob--W and @xeenon indicated that they'd like to see more details about the nature of the request. What is the desired outcome? Do you have a rough API shape or feature set that you'd like to see?

**Comment by yankovichv** [comment]
@dotproto, our goal is to achieve full functionality for our tab manager.

Chrome's tab manager displays the recording status. We are unable to do this with the chrome.tabs permission.

I think the `recording: true/false` property on the tab object would be the best solution.

**Comment by yankovichv** [comment]
@oliverdunk, what do you think about the `mute site` feature for extensions in Chrome?

**Comment by oliverdunk** [comment]
> @oliverdunk, what do you think about the `mute site` feature for extensions in Chrome?

I haven't spent too much time looking at this, but I'm wondering if the [chrome.contentSettings API](https://developer.chrome.com/docs/extensions/reference/contentSettings/) would be a good home for this. It doesn't feel like a high-priority new API but it may be easy enough to add to something existing.

**Comment by yankovichv** [comment]
@oliverdunk , wow, sounds good. If only there were events 🤤
